tests/optimizers/test_general_optimizer.py

```diff
@@ -52,6 +67,21 @@ def test_invalid_k_or_p_values(options):
         GeneralOptimizerPSO(5, 2, options, Ring())
 
 
```

Comment:
Great tests! To be honest, the `GeneralOptimizer` can make everything [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). The reason we will still keep the GlobalBest and LocalBest is for tradition&mdash; they're the commonly-known PSO algorithms, and having them exposed to a first-time user is good.